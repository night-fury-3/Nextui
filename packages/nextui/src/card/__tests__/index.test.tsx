import React from 'react';
import { mount, render } from 'enzyme';
import Card from '../index';

describe('Card', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Card>card</Card>);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should support hoverable', () => {
    const wrapper = render(
      <div>
        <Card hoverable>card</Card>
      </div>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should support clikable', () => {
    const wrapper = render(
      <div>
        <Card clickable>card</Card>
      </div>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should support custom css', () => {
    const wrapper = render(
      <div>
        <Card css={{ bg: '$red400' }}>card</Card>
      </div>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should support card types', () => {
    const wrapper = mount(
      <div>
        <Card color="default">card</Card>
        <Card color="primary">card</Card>
        <Card color="success">card</Card>
        <Card color="secondary">card</Card>
        <Card color="warning">card</Card>
        <Card color="gradient">card</Card>
      </div>
    );
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should render correctly when nested', () => {
    const wrapper = render(
      <Card>
        <Card shadow>
          <Card>card</Card>
        </Card>
      </Card>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('the component Card.Content should be injected automatically', () => {
    const card = mount(<Card>test-value</Card>);
    const content = mount(
      <Card>
        <Card.Body>test-value</Card.Body>
      </Card>
    );
    expect(card.html()).toEqual(content.html());
  });
});
