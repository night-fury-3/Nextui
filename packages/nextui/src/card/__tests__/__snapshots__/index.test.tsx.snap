// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly when nested 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "nextui-c-nVPDe nextui-c-nVPDe-kQghEc-color-default nextui-c-nVPDe-iVCgFB-shadow-true nextui-c-nVPDe-bVtBym-bordered-false nextui-c-nVPDe-bzEnZV-borderWeight-normal nextui-c-nVPDe-iNKybL-animated-true nextui-c-nVPDe-bxaxLq-cover-false nextui-c-nVPDe-kQghEc-cv",
      "role": "section",
      "tabindex": "-1",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "nextui-c-gJBdwd",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "nextui-c-nVPDe nextui-c-nVPDe-kQghEc-color-default nextui-c-nVPDe-iVCgFB-shadow-true nextui-c-nVPDe-bVtBym-bordered-false nextui-c-nVPDe-bzEnZV-borderWeight-normal nextui-c-nVPDe-iNKybL-animated-true nextui-c-nVPDe-bxaxLq-cover-false nextui-c-nVPDe-kQghEc-cv",
              "role": "section",
              "tabindex": "-1",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "nextui-c-gJBdwd",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "class": "nextui-c-nVPDe nextui-c-nVPDe-kQghEc-color-default nextui-c-nVPDe-iVCgFB-shadow-true nextui-c-nVPDe-bVtBym-bordered-false nextui-c-nVPDe-bzEnZV-borderWeight-normal nextui-c-nVPDe-iNKybL-animated-true nextui-c-nVPDe-bxaxLq-cover-false nextui-c-nVPDe-kQghEc-cv",
                      "role": "section",
                      "tabindex": "-1",
                    },
                    "children": Array [
                      Node {
                        "attribs": Object {
                          "class": "nextui-c-gJBdwd",
                        },
                        "children": Array [
                          Node {
                            "data": "card",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "role": undefined,
                      "tabindex": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "role": undefined,
                      "tabindex": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "role": undefined,
              "tabindex": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "role": undefined,
              "tabindex": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Card should support clikable 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "attribs": Object {
          "class": "nextui-c-nVPDe nextui-c-nVPDe-kQghEc-color-default nextui-c-nVPDe-iVCgFB-shadow-true nextui-c-nVPDe-bVtBym-bordered-false nextui-c-nVPDe-bzEnZV-borderWeight-normal nextui-c-nVPDe-iNKybL-animated-true nextui-c-nVPDe-lkBgbl-clickable-true nextui-c-nVPDe-bxaxLq-cover-false nextui-c-nVPDe-kQghEc-cv nextui-c-nVPDe-expcIv-cv",
          "role": "button",
          "tabindex": "0",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "nextui-c-gJBdwd",
            },
            "children": Array [
              Node {
                "data": "card",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
          "tabindex": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
          "tabindex": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Card should support custom css 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "attribs": Object {
          "class": "nextui-c-nVPDe nextui-c-nVPDe-kQghEc-color-default nextui-c-nVPDe-iVCgFB-shadow-true nextui-c-nVPDe-bVtBym-bordered-false nextui-c-nVPDe-bzEnZV-borderWeight-normal nextui-c-nVPDe-iNKybL-animated-true nextui-c-nVPDe-bxaxLq-cover-false nextui-c-nVPDe-kQghEc-cv nextui-c-nVPDe-ikCXDdt-css",
          "role": "section",
          "tabindex": "-1",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "nextui-c-gJBdwd",
            },
            "children": Array [
              Node {
                "data": "card",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
          "tabindex": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
          "tabindex": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Card should support hoverable 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "attribs": Object {
          "class": "nextui-c-nVPDe nextui-c-nVPDe-kQghEc-color-default nextui-c-nVPDe-iVCgFB-shadow-true nextui-c-nVPDe-bVtBym-bordered-false nextui-c-nVPDe-bzEnZV-borderWeight-normal nextui-c-nVPDe-iNKybL-animated-true nextui-c-nVPDe-jQJOMy-hoverable-true nextui-c-nVPDe-bxaxLq-cover-false nextui-c-nVPDe-kQghEc-cv",
          "role": "section",
          "tabindex": "-1",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "nextui-c-gJBdwd",
            },
            "children": Array [
              Node {
                "data": "card",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
          "tabindex": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
          "tabindex": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
